spring-boot-project/spring-boot-docs/src/docs/asciidoc/production-ready-features.adoc

```diff
@@ -2387,6 +2388,97 @@ For more details refer to {spring-kafka-docs}#micrometer-native[Micrometer Nativ
 
 
 
+[[production-ready-metrics-mongodb]]
+==== MongoDB Metrics
+
+===== Command Metrics
+Auto-configuration will register a `MongoMetricsCommandListener` for the auto-configured MongoClient.
+
+A timer metric with the name `mongodb.driver.commands` is created for each command issued to the underlying MongoDB driver.
+Each metric is tagged with the following information by default:
+|===
+| Tag | Description
+
+| `command`
+| Name of the command issued
+
+| `cluster.id`
+| Identifier of the cluster the command was sent to
+
+| `server.address`
+| Address of the server the command was sent to
+
+| `status`
+| Outcome of the command - one of (`SUCCESS`, `FAILED`)
+|===
+
+You can replace the default metric tags by providing a `MongoMetricsCommandTagsProvider` bean, as shown in the following example:
+
+[source,java,pending-extract=true,indent=0]
+----
+	@Bean
+	MongoMetricsCommandTagsProvider mongoMetricsCommandTagsProvider() {
+		return new MyCustomMongoMetricsCommandTagsProvider();
+	}
+----
+
+If you want to disable the auto-configured command metrics, you can set the following property:
+
+[source,yaml,indent=0,configprops,configblocks]
+----
+	management:
+	  metrics:
+	    mongo:
+	      commandlistener:
+	        enabled: false
+----
+
+===== Connection Pool Metrics
+Auto-configuration will register a `MongoMetricsConnectionPoolListener` for the auto-configured MongoClient.
+
+The following gauge metrics are created for the connection pool:
+
+* `mongodb.driver.pool.size` that reports the current size of the connection pool, including idle and and in-use members
+* `mongodb.driver.pool.checkedout` that reports the count of connections that are currently in use
+* `mongodb.driver.pool.waitqueuesize` that reports the current size of the wait queue for a connection from the pool
+
+NOTE: The `waitqueuesize` metric was https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#why-are-waitqueuesize-and-waitqueuemultiple-deprecated[removed] in MongoDB 4.x
```

Comment:
@jonatan-ivanov I went ahead and added the info about the `waitqueuesize` deprecation here as well.

Reply:
The documentation that's linked to here doesn't seem to be directly related to the metric. The documentation is talking about the removal of a configuration setting rather than a metric for observing the current wait queue size. Given that Micrometer's `MongoMetricsConnectionPoolListener` provides the metric without any reliance of Mongo Driver APIs, I don't think there's any need to mention the deprecation here.

Reply:
Fair enough. Ideally those docs would live in Micrometer anyways and at best this would just point to their docs. I will remove.